linera-rpc/src/grpc/client.rs

```diff
@@ -316,6 +316,34 @@ impl ValidatorNode for GrpcClient {
         Ok(client_delegate!(self, download_certificate, hash)?.try_into()?)
     }
 
+    #[instrument(target = "grpc_client", skip_all, err, fields(address = self.address))]
+    async fn download_certificates(
+        &self,
+        hashes: Vec<CryptoHash>,
+    ) -> Result<Vec<Certificate>, NodeError> {
+        let mut missing_hashes = hashes;
+        let mut certs_collected = Vec::with_capacity(missing_hashes.len());
+        loop {
+            // Macro doesn't compile if we pass `missing_hashes.clone()` directly to `client_delegate!`.
+            let missing = missing_hashes.clone();
+            let mut received: Vec<Certificate> =
+                client_delegate!(self, download_certificates, missing)?.try_into()?;
```

Comment:
This is probably not relevant any time soon, but in the long run we may want to already limit the number of hashes we request here. In extreme cases, even the _request_ might exceed the message size limit.

Reply:
I can do it right away. It should not be a difficult Change.

Reply:
Lifted to an issue https://github.com/linera-io/linera-protocol/issues/2659